telegram/_bot.py

```diff
@@ -1889,7 +1923,7 @@ async def send_media_group(
                 item_to_get_caption.caption_entities = parse_sequence_arg(caption_entities)
 
             # copy the list (just the references) to avoid mutating the original list
-            media = media[:]
+            media = list(media)
```

Comment:
If no caption is passed and user passes a tuple as `media`, then it will remain a tuple in `data` dict. Is it OK? I looked at some protected methods (what they do with items in `data` before sending to Telegram) and I'm not sure a tuple will be handled correctly. For instance, `RequestParameter.from_input()` has an `isinstance(value), list` check.

If I'm correct, then maybe `media = list(media)` should be moved up in the code, outside of `if caption` clause.

Reply:
As mentioned in the original issue, json module automatically convert tuples to json arrays which will also work. You can see that `RequestParameter.from_input()` has changed its check from `isinstance(value, list)` to  `if not isinstance(value, str) and isinstance(value, Sequence):`

Reply:
Right, I was looking at the old file in my local branch 🤦🏻